src/server.rs

```diff
@@ -55,7 +57,22 @@ async fn create_server() -> axum::Server<AddrIncoming, IntoMakeService<axum::Rou
         )
         .with_state(conf.leptos_options.clone());
 
-    let server = axum::Server::bind(&conf.leptos_options.site_addr).serve(app.into_make_service());
+    let site_addr = leptos_site_addr
+        .map(|s| s.parse::<SocketAddr>())
+        .map(|s| match s {
+            Ok(site_addr) => site_addr,
+            Err(err) => {
+                tracing::error!(
+                    "{}, defaulting to: {}",
+                    err,
+                    conf.leptos_options.site_addr.to_string()
+                );
+                conf.leptos_options.site_addr
+            }
+        })
+        .unwrap_or(conf.leptos_options.site_addr);
```

Comment:
Instead of doing this, this line can be made a function of its own taking cli args and returning the effective leptos_options:

https://github.com/juspay/nix-browser/blob/f1d0a4af2a41a18c02f5cba85e9e8637247ec157/src/server.rs#L33

Reply:
```rust
fn get_leptos_options(args: &Args) -> LeptosOptions {
  // TODO: handle args.site_addr
  get_configuration(None).await.unwrap().conf;
}
```

Reply:
sure, i'll update this

Reply:
i have added `get_leptos_options` which returns the updated `LeptosOptions`